### DISCUSSION TITLE
Spatie Media Column

### DISCUSSION THREAD

**Discussion by TheROPFather** [Laravel-Backpack/community-forum#557] [Q&A (Help)] [answered]
Trying to use the new integration with Spatie Media Library and have run into this issue.

When trying to show the column in the list like so 

`CRUD::column('banner')->type('image')->withMedia();`

Nothing happens, I'm thinking it's because there is no field in the model called banner. When using Spatie Media Library the database doesn't have column names on the model, so I'm not sure how I can access this.

I have also tried 

`CRUD::column('banner')->type('image')->withMedia(['collection' => 'banner']);`

Not sure how to proceed.

**Comment by TheROPFather** [comment]
As an update to this.

I added the field to my edit, and after that the media DOES show. So it appears to only work with files that have been uploaded using the `withMedia()` field.

If you have existing associated media it will not show or if you associate the media outside of the CRUD. I'm not sure if this is intentional, but I think it should show any media.

**Reply by TheROPFather** [reply]
Also may be worth noting as I work through this, the old image field has no way to use the original filename. This makes sense when not using medialibrary but if you are, the filename collision is a non-issue and there are situations where maintaining the original filename are important.

**Comment by pxpm** [comment, marked as answer]
Thanks for the suggestions. 

I will try to reproduce the issues you mentioned but I guess it's because previous media does not have the custom properties required to work with uploaders. 

We need to save some additional information in the custom properties of the media, and we use that information for display/saving purposes. 

I don't think we can do something about it (in the current implementation), just probably document how to do it and developer could run a database query to update their media accordingly to the fields where it's used. 

Maybe expose a different API in case there is no "custom properties" set by CRUD, you could set them in the column/field to work with media saved outside crud/previous media.

https://github.com/Laravel-Backpack/medialibrary-uploaders#custom-properties

If you are not using it inside repeatable the only custom attribute you need to set in the media custom properties is the `name`, that should match the uploader name AKA the field name. 

Do you have any suggestion regarding this ? 

Cheers

**Reply by TheROPFather** [reply]
I was unaware that you needed custom properties set. In my situation I can just add the `name` custom property manually as you advised. 

I think a sane default would be if a collection is passed like I have above, then the column could perform its check and if no custom property for name is found then attempt just using the collection name as a fallback. 

For now this solution is fine for me, I wonder if you would also consider allowing maintaining the original image name - I'm aware this is a separate issue and I will open an issue in the correct place if you think that would be helpful. 

Thanks for the response. 

Love the work you guys put in on this, it's a HUGE help for my project - allowed me to get rid of my own custom implementation :)

**Reply by pxpm** [reply]
Please open issues with what you think we could do different/better or just new ideas. If you find a bug we would also love to know 😅 

We care for you guys, thanks for the kind words 🙏 

Cheers
